extension-manager-integration-test-java/src/main/java/com/exasol/extensionmanager/itest/PreviousExtensionVersion.java

```diff
@@ -28,15 +30,21 @@ private PreviousExtensionVersion(final Builder builder) {
         this.extensionFileName = Objects.requireNonNull(builder.extensionFileName, "extensionFileName");
         this.currentVersion = Objects.requireNonNull(builder.currentVersion, "currentVersion");
         this.previousVersion = Objects.requireNonNull(builder.previousVersion, "previousVersion");
-        this.adapterFileName = Objects.requireNonNull(builder.adapterFileName, "adapterFileName");
+        this.adapterFileName = builder.adapterFileName;
```

Comment:
Long-term I would replace this with bucketfs_files and maybe make a specialization for java virtual schema. I added a diagram, what kind of extensions we can expect.
![extension_hiearchy drawio](https://github.com/exasol/extension-manager/assets/5796845/dc2b8d2a-23f8-40d4-94e3-31ba0a1f1f88)